File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int N = nums.size() - 1;
        int left = 0;
        int right = N/2;
        int mid = left + right;
        while (left < right){
            mid += 1 - mid%2;
            if(nums[mid] == nums[mid-1]){
                    left = mid/2 + 1;
            }
            else{
                right = mid/2;
            }
            mid = left + right;
        }
        return nums[left*2];
    }
};

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        N = len(nums) - 1
        left = 0
        right = N//2
        mid = left + right
        while mid < N:
            mid -= mid%2
            if nums[mid] != nums[mid+1]:
                if mid == 0 or nums[mid] != nums[mid-1]:
                    return nums[mid]
                right = mid//2 - 1
            else:
                left = mid//2 + 1
            mid = left + right
        return nums[N]